src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -58,44 +72,116 @@ public override async Task<LSP.CompletionItem[]> HandleRequestAsync(LSP.Completi
             if (list == null)
             {
                 return Array.Empty<LSP.CompletionItem>();
             }
 
             var lspVSClientCapability = context.ClientCapabilities?.HasVisualStudioLspCapability() == true;
 
-            return list.Items.Select(item => CreateLSPCompletionItem(request, item, lspVSClientCapability)).ToArray();
+            return list.Items.Select(item => CreateLSPCompletionItem(request, item, lspVSClientCapability, completionTrigger)).ToArray();
 
             // local functions
-            static LSP.CompletionItem CreateLSPCompletionItem(LSP.CompletionParams request, CompletionItem item, bool useVSCompletionItem)
+            static LSP.CompletionItem CreateLSPCompletionItem(
+                LSP.CompletionParams request,
+                CompletionItem item,
+                bool useVSCompletionItem,
+                CompletionTrigger completionTrigger)
             {
                 if (useVSCompletionItem)
                 {
-                    var vsCompletionItem = CreateCompletionItem<LSP.VSCompletionItem>(request, item);
+                    var vsCompletionItem = CreateCompletionItem<LSP.VSCompletionItem>(request, item, completionTrigger);
                     vsCompletionItem.Icon = new ImageElement(item.Tags.GetFirstGlyph().GetImageId());
                     return vsCompletionItem;
                 }
                 else
                 {
-                    var roslynCompletionItem = CreateCompletionItem<RoslynCompletionItem>(request, item);
+                    var roslynCompletionItem = CreateCompletionItem<RoslynCompletionItem>(request, item, completionTrigger);
                     roslynCompletionItem.Tags = item.Tags.ToArray();
                     return roslynCompletionItem;
                 }
             }
 
-            static TCompletionItem CreateCompletionItem<TCompletionItem>(LSP.CompletionParams request, CompletionItem item) where TCompletionItem : LSP.CompletionItem, new()
-                => new TCompletionItem
+            static TCompletionItem CreateCompletionItem<TCompletionItem>(
+                LSP.CompletionParams request,
+                CompletionItem item,
+                CompletionTrigger completionTrigger) where TCompletionItem : LSP.CompletionItem, new()
+            {
+                var completeDisplayText = item.DisplayTextPrefix + item.DisplayText + item.DisplayTextSuffix;
+                var completionItem = new TCompletionItem
                 {
-                    Label = item.DisplayTextPrefix + item.DisplayText + item.DisplayTextSuffix,
-                    InsertText = item.Properties.ContainsKey("InsertionText") ? item.Properties["InsertionText"] : item.DisplayText,
+                    Label = completeDisplayText,
+                    InsertText = item.Properties.ContainsKey("InsertionText") ? item.Properties["InsertionText"] : completeDisplayText,
```

Comment:
Changed from `item.DisplayText` to `completeDisplayText` to address the `=` case (discussed in https://github.com/dotnet/roslyn/pull/45099#discussion_r439699963).